src/plugins/mentionRoleIcons/index.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2024 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import definePlugin from "@utils/types";
+import { useState, useEffect, RestAPI } from "@webpack/common";
+import { Devs } from "@utils/constants";
+
+import "./styles.css";
+
+export default definePlugin({
+    name: "MentionRoleIcons",
+    description: "Shows roles icons inside role mentions",
+    authors: [Devs.SerStars],
+
+    patches: [{
+        find: ".ROLE_MENTION)",
+        replacement: {
+            match: /function \i\(\i\){let{roleColor:.+?,roleId:(\i),.+?roleName:\i,guildId:(\i).+?}=\i,{analyticsLocations:.+\.ROLE_MENTION\),.+\i&&\(0,\i\.jsx\)\(\i\.RoleDot,{color:\(0,\i\.\i\).+?\.roleDot,background:.+?(1|0)}\),\i/,
+            replace: "$&,$self.AddIcon($1, $2)"
+        }
```

Comment:
```suggestion
            find: ".ROLE_MENTION)",
            replacement: {
                match: /\.\.\.\i,children:\[.{0,100}\]/,
                replace: "$&.push($self.AddIcon(arguments[0]))"
            }
```
makes the patch a bit less hellish